docs/recipes/install/rocky9/x86_64/openchami/slurm/recipe.sh

```diff
@@ -0,0 +1,633 @@
+
+
+# ------------------------------------------
+# Enable OpenCHAMI Repositories and Support 
+# ------------------------------------------
+dnf -y install dnsmasq podman buildah git openssl ansible-core nfs-utils tcpdump tftp curl
+rpm --import https://github.com/OpenCHAMI/release/releases/download/v0.0.26/public_gpg_key.asc
+dnf -y install https://github.com/OpenCHAMI/release/releases/download/v0.0.26/openchami-0.0.26.rpm
+dnf -y install openchami
+
+# --------------------------------
+# Creating OpenCHAMI certificates 
+# --------------------------------
+echo "${sms_ip} ${sms_name}" | tee -a /etc/hosts > /dev/null
+systemctl start step-ca
+podman run --rm --network openchami-cert-internal docker.io/curlimages/curl -sk https://step-ca:9000/roots.pem | tee /etc/pki/ca-trust/source/anchors/ochami.pem
+update-ca-trust
+systemctl restart acme-deploy
+systemctl restart haproxy
+
+
+# -----------------------------------------------
+# Complete basic OpenCHAMI setup for master node 
+# -----------------------------------------------
+mkdir -p /opt/ohpc/admin
+mkdir -p /opt/ohpc/admin/data/oci
+cat > /etc/containers/systemd/registry.container <<EOF
+# registry.container
+[Unit]
+Description=Image OCI Registry
+After=network-online.target
+Requires=network-online.target
+[Container]
+ContainerName=registry
+HostName=registry
+Image=docker.io/library/registry:latest
+Volume=/opt/ohpc/admin/data/oci:/var/lib/registry:Z
+PublishPort=5000:5000
+[Service]
+TimeoutStartSec=0
+Restart=always
+[Install]
+WantedBy=multi-user.target
+EOF
+mkdir -p /opt/ohpc/admin/data/s3
+cat > /etc/containers/systemd/minio.container <<EOF
+[Unit]
+Description=Minio S3
+After=local-fs.target network-online.target
+Wants=local-fs.target network-online.target
+[Container]
+ContainerName=minio-server
+Image=docker.io/minio/minio:latest
+# Volumes
+Volume=/opt/ohpc/admin/data/s3:/data:Z
+# Ports
+PublishPort=9000:9000
+PublishPort=9091:9001
+# Environemnt Variables
+Environment=MINIO_ROOT_USER=admin
+Environment=MINIO_ROOT_PASSWORD=password1234
+# Command to run in container
+Exec=server /data --console-address :9001
+[Service]
+Restart=always
+[Install]
+WantedBy=multi-user.target
+EOF
+
+# Define local domainname, deployment protocol and dns
+systemctl daemon-reload
+systemctl start minio.service
+systemctl start registry.service
+
+# ---------------
+# Configure DHCP 
+# ---------------
+cat > /etc/openchami/configs/coredhcp.yaml <<EOF
+server4:
+ plugins:
+    - server_id: ${sms_ip} 
+    - router: ${internal_network%.*}.254 
+EOF
+
+
+# -------------------
+# Starting OpenCHAMI 
+# -------------------
+systemctl start openchami.target
+
+
+# -------------------------
+# OpenCHAMI Client Install 
+# -------------------------
+latest_release_url=$(curl -s https://api.github.com/repos/OpenCHAMI/ochami/releases/latest | jq -r '.assets[] | select(.name | endswith("amd64.rpm")) | .browser_download_url')
+curl -L "${latest_release_url}" -o ochami.rpm
+dnf install -y ./ochami.rpm
```

Comment:
Not really necessary to first do a curl and then a dnf install. You could pass the URL directly to dnf.